### DISCUSSION TITLE
How can I get the Pool object?

### DISCUSSION THREAD

**Discussion by cpouldev** [redis/go-redis#2497] [Q&A]
Hey all

I'm using redisearch module (along with the relevant go library) and I want to [share the connection pool instance](https://pkg.go.dev/github.com/RediSearch/redisearch-go/redisearch#NewClientFromPool) from the go-redis with it.

I dived into the source code and I couldn't find any exposed getter or something relevant. 

Any help / recommendation will be appreciated. 

----
UPDATE:

**I just realized that go-redis and redisgo are different, therefore the Pool implementation is different too.**

**Comment by monkey92t** [comment]
go-redis allows access to network connection objects, while *Conn objects are still managed by go-redis/pool.
```go
rdb := redis.NewClient(...)
conn := rdb.Conn()
```

However, it's important to note that, as you mentioned, the go-redis connection pool is closed and similar to database/sql, we don't advocate for using individual network connections.

But if you need to perform some special work, you'll need to accept the risks involved with doing so.

**Comment by Jun10ng** [comment]
hi @cpouldev did u find any solution? I have the same question.

**Reply by cpouldev** [reply]
Hi, as I remember unfortunately no

**Reply by Jun10ng** [reply]
I decided to use redisHook function to operate redis conn.
